File: core/src/main/java/com/triangleleft/flashcards/ui/common/presenter/IPresenter.java
package com.triangleleft.flashcards.ui.common.presenter;

import android.support.annotation.Nullable;

import com.triangleleft.flashcards.ui.ViewState;
import com.triangleleft.flashcards.ui.common.view.IView;

public interface IPresenter<View extends IView, VS extends ViewState> {
    @Deprecated
    void onCreate();

    void onCreate(@Nullable VS savedViewState);

    @Deprecated
    void onBind(View view);

    void onRebind(View view);

    void onUnbind();

    void onDestroy();

    View getView();

    VS getViewState();
}

File: core/src/main/kotlin/com/triangleleft/flashcards/ui/common/IPresenter.kt
package com.triangleleft.flashcards.ui.common.presenter

import com.triangleleft.flashcards.ui.ViewState
import com.triangleleft.flashcards.ui.common.view.IView

interface IPresenter<View : IView<*, *>, VS : ViewState> {

    fun onCreate(savedViewState: VS?)

    fun onViewCreated()

    fun onBind(view: View)

    fun onUnbind()

    var view: View?

    fun onDestroy()
}
